### PR TITLE
Ignore media features in a range context

### LINKED ISSUE #1728: media-feature-name-case needs to account (and have tests) for range context
Ref: https://github.com/stylelint/stylelint/issues/1722#issuecomment-235635385

[postcss-media-query-parser](https://github.com/dryoma/postcss-media-query-parser) does not support media feature in a range context. We should probably ignore media features in a range context for `7.1` and then, possibly, add support in `future-major` if the spec matures.

Ignoring range context within these rules might create a odd user experience though as we have two rules (`media-feature-range-operator-space-*`) specifically for range context. I can't see another way around it though. Ideas?

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #1728

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@jeddy3 LGTM, we need create issue in `postcss-media-query-parser` about support range operators and create issue in `stylelint` repo with label `future-major` about remove this `ignore`. Agree?

**Comment by evilebottnawi** [comment]
@jeddy3 After merge this i will add tests to new rule `https://github.com/stylelint/stylelint/pull/1711`

**Comment by jeddy3** [comment]
>  create issue in stylelint repo with label future-major about remove this ignore

SGTM.

>  we need create issue in postcss-media-query-parser about support range operators

That's already [been discussed](https://github.com/dryoma/postcss-media-query-parser/issues/1#issuecomment-233054873), right? If we want to support range context for these rules we need to write our own (or wait for someone else to do it) [range context parser](https://www.w3.org/TR/mediaqueries-4/#mq-range-context). We can then use `postcss-media-query-parser` to parse the whole *media query list*, but use the *range context parser* on "media feature expressions" that are in a range context.

**Comment by evilebottnawi** [comment]
@jeddy3 I'm not understand why it can not be implement in the `postcss-media-query-parser`?

**Comment by jeddy3** [comment]
>  I'm not understand why it can not be implement in the postcss-media-query-parser?

I think dryoma [intends](https://github.com/dryoma/postcss-media-query-parser/issues/2#issuecomment-233118704) `postcss-media-query-parser` for the 2013 RC Media Query spec. Range context is in an Editors Draft spec. I understand the appeal of creating a separate experimental parser for that feature, as we just don't know if it's going to change or not.

**Comment by evilebottnawi** [comment]
@jeddy3 maybe to try to convince to add parsing range operators? As experimental

File: src/utils/isRangeContextMediaFeature.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * Check whether a media feature is a range context one
+ *
+ * @param {string} media feature
+ * @return {boolean} If `true`, media feature is a range context one
+ */
+export default function (mediaFeature) {
+  return (
+    mediaFeature.indexOf("=") !== -1
+    || mediaFeature.indexOf("<") !== -1
+    || mediaFeature.indexOf(">") !== -1
```
Comment (davidtheclark) [inline src/utils/isRangeContextMediaFeature.js]: This works fine. If anybody wants to dig into micro-optimizations, we could do all this in one loop, or see if a regexp is faster.

**Comment by davidtheclark** [comment]
LGTM

**Comment by davidtheclark** [comment]
Let's add a changelog note then merge.

**Comment by jeddy3** [comment]
> Let's add a changelog note then merge.

This rule isn't released yet, so this changed is covered in the `Added: ...` changelog item.

**Pull request merged**